test/t/conftest.py

```diff
@@ -305,6 +305,9 @@ def bash(request) -> pexpect.spawn:
             )
             if match:
                 cmd = match.group(1)
+        if marker and marker.kwargs and marker.kwargs.get("require_cmd", False):
+            if not is_bash_type(bash, cmd):
+                pytest.skip("Command not found")
```

Comment:
This looks like it's useful no matter what happens to the rest of this PR, so it would have been better to place it in a PR of its own, for cleanliness and if we happen to need to revert the rest for some reason later. But I don't mind keeping it in here this time.

Reply:
OK. I'll separate the PR.

Reply:
This was merged in PR #547.